coral-component-table/src/scripts/Table.js

```diff
@@ -1174,6 +1180,126 @@ const Table = Decorator(class extends BaseComponent(HTMLTableElement) {
+      const dragData = {
+        placeholder: placeholder,
+        index: index,
+        oldBefore: oldBefore,
+        // Backup styles to restore them later
+        style: {
+          row: style
+        }
+      };
+      row.dragAction._dragData = dragData;
+    }
+  }  
+
+  /** @private */
+  _onRowDragOnKeySpace(event) {
+    const dragElement = event.detail.dragElement;
+    const dragData = dragElement.dragAction._dragData;
+
+    dragData.style.cells = [];
+    getCells(dragElement).forEach((cell) => {
+      // Backup styles to restore them later
+      dragData.style.cells.push(cell.getAttribute('style'));
+      // Cells will shrink otherwise
+      cell.style.width = window.getComputedStyle(cell).width;
+    });
+  }  
+
+  /** @private */
+  _onRowDragOverOnKeyArrowDown(event) {
+    const table = this;
+    const body = table.body;
+    const dragElement = event.detail.dragElement;
+    const items = getRows([body]);
+    const index = getIndexOf(dragElement);
+    const dragData = dragElement.dragAction._dragData;
+
+    // We cannot rely on :focus since the row is being moved in the dom while dnd
+    dragElement.classList.add('is-focused');
+
+    if (dragElement === items[items.length - 1]) {
+      for (let position = 0; position < items.length - 1; position++) {
+        body.appendChild(items[position]);
+      }
+      body.insertBefore(items[0], items[items.length - 2].nextElementSibling);
+    } else {
+      body.insertBefore(items[index + 1], items[index]);
+    }
+
+    // Restore specific styling
+    dragElement.setAttribute('style', dragData.style.row || '');
+    getCells(dragElement).forEach((cell, i) => {
+      cell.setAttribute('style', dragData.style.cells[i] || '');
+    });  
+  }
+
+  /** @private */
+  _onRowDragOverOnKeyArrowUp(event) {
+    const table = this;
+    const body = table.body;
+    const dragElement = event.detail.dragElement;
+    const items = getRows([body]);
+    const index = getIndexOf(dragElement);
+    const dragData = dragElement.dragAction._dragData;
+
+    // We cannot rely on :focus since the row is being moved in the dom while dnd
+    dragElement.classList.add('is-focused');
+
+    if (dragElement === items[0]) {
+      for (let position = 0; position < items.length - 2; position++) {
+        body.insertBefore(items[position + 1], items[0]);
+      }
+      body.insertBefore(items[items.length - 1], items[1]);
+    } else {
+      body.insertBefore(items[index - 1], items[index].nextElementSibling);
+    }
+
+    // Restore specific styling
+    dragElement.setAttribute('style', dragData.style.row || '');
+    getCells(dragElement).forEach((cell, i) => {
+      cell.setAttribute('style', dragData.style.cells[i] || '');
+    });  
+  }
+  
+  /** @private */
+  _onRowDragOverOnKeyEnter(event) {
+    const table = this;
+    const dragElement = event.detail.dragElement;
+    dragElement.dragAction.destroy();
```

Comment:
```suggestion
    const dragAction = dragElement.dragAction;
    const handle = dragAction.handle;
    const span = handle && handle.closest('span[role="application"]');

    // If keyboard dragging, do nothing, and wait for event to be triggered from DragAction.
    if (dragAction._isKeyboardDrag) {
      dragAction._isKeyboardDrag = undefined;
      return;
    }

    if (handle && handle.getAttribute('is') === 'coral-button') {
      handle.selected = false;
      handle.removeAttribute('aria-pressed');
    }

    dragAction.destroy();
```

Reply:
We need to dispatch `coral-table:reorder` events:

https://github.com/adobe/coral-spectrum/pull/242/files#diff-bdeeff66476e39839ee40448b1099869cead00f833526b0ad65bb8f6034365b7R1149-R1174